doc/usage/extensions/intersphinx.rst

```diff
@@ -148,6 +148,25 @@ linking:
       exception is raised if the server has not issued a response for timeout
       seconds.
 
+.. confval:: intersphinx_disabled_domains
+
+   .. versionadded:: 4.2
+
+   A list of strings being the name of a domain, or the special name ``all``.
+   When a cross-reference without an explicit inventory specification is being
+   resolved by intersphinx, skip resolution if either the domain of the
+   cross-reference is in this list or the special name ``all`` is in the list.
+
+   For example, with ``intersphinx_disabled_domains = ['std']`` a cross-reference
+   ``:doc:`installation``` will not be attempted to be resolved by intersphinx, but
+   ``:doc:`otherbook:installation``` will be attempted to be resolved in the
```

Comment:
I think explicit references should be also disabled. It would be replaced by https://github.com/sphinx-doc/sphinx/pull/9062#issuecomment-879861369.

Reply:
Note that I wrote this PR as if https://github.com/sphinx-doc/sphinx/pull/9062#issuecomment-879861369 didn't exist. My plan is to have the functionality change incrementally:
1. This PR, for which this part is as intended: the user must intentionally write something intersphinx-specific to activate it for the disabled domains.
2. Then an updated #9062 which uses the functionality in this PR. By then the old style of explicit intersphinx references, as the one you quoted here, can be deprecated/removed.
3. The whole domain delegation part.

So in short, yes, once an updated #9062 has been merged, I don't mind this option disabling the explicit ones as well, but I don't think it is harmful if they are not.